pkg/plugins/policies/core/rules/common/resolvetargetref.go

```diff
@@ -80,14 +85,40 @@ func ResolveTargetRef(targetRef common_api.TargetRef, tMeta core_model.ResourceM
 	}
 
 	ri := core_model.TargetRefToResourceIdentifier(tMeta, targetRef)
-	if resource := reader.Get(rtype, ri); resource != nil {
-		return []*ResourceSection{{
-			Resource:    resource,
-			SectionName: targetRef.SectionName,
-		}}
+	resolvedResource := reader.Get(rtype, ri)
+	if resolvedResource == nil {
+		return nil
 	}
 
-	return nil
+	if resourceWithPorts, ok := resolvedResource.(ResourceWithPorts); ok && targetRef.SectionName != "" {
+		for _, port := range resourceWithPorts.GetPorts() {
+			if port.GetName() == targetRef.SectionName {
+				return []*ResourceSection{{
+					Resource:    resolvedResource,
+					SectionName: port.GetName(),
+				}}
+			}
+			if targetRefPort, ok := tryParsePort(targetRef.SectionName); ok && port.GetName() == "" && port.GetValue() == targetRefPort {
```

Comment:
I went through the code and we set port value as a port name if there is no name. Won't it affect matching here?

Reply:
> we set port value as a port name if there is no name

hmm, could you please share code where it happens

Reply:
One for universal other for k8s
https://github.com/kumahq/kuma/blob/master/pkg/core/resources/apis/meshservice/generate/generator.go#L90
https://github.com/kumahq/kuma/blob/master/pkg/plugins/runtime/k8s/controllers/meshservice_controller.go#L397

Reply:
I don't think so, since MeshMultizoneService is created manually and we only update status object. It can be defined by the user: https://kuma.io/docs/2.9.x/networking/meshmultizoneservice/#ports

Reply:
right, so the second condition
```go
if targetRefPort, ok := tryParsePort(targetRef.SectionName); ok && port.GetName() == "" && port.GetValue() == targetRefPort {
```
makes sense